app/apps/server/bridges/rooms.js

```diff
@@ -84,6 +85,66 @@ export class AppRoomBridge {
 		return this.orch.getConverters().get('users').convertById(room.u._id);
 	}
 
+	async getMessages(roomId, appId) {
+		this.orch.debugLog(`The App ${ appId } is getting the room's messages by room id: "${ roomId }"`);
+
+		const room = Rooms.findOneById(roomId);
+		if (!room) {
+			throw new Error(`Room could not be found by id ${ roomId }`);
+		}
+
+		const orcha = this.orch;
+		const totalCount = await MessagesRaw.countVisibleByRoomId(roomId);
+
+		return {
+			[Symbol.asyncIterator]: function getMessageByRoomIdIterator() {
+				let exhausted = false;
+				let currentIndex = 0;
+
+				return {
+					async next() {
+						// currentIndex will always be 0 based whereas count is always 1 based.
+						// so we check if it is equal to or greater than the total count
+						// and if so, then we are done with this iteration
+						if (currentIndex >= totalCount || exhausted) {
+							return {
+								value: undefined,
+								done: true,
+							};
+						}
+
+						// We skip the current index because if it is the first time, then we skip zero
+						// if it is the second time then we skip one, and so on and so forth
+						const options = { limit: 1, sort: { ts: 1 }, skip: currentIndex };
+						const result = await MessagesRaw.findVisibleByRoomId(roomId, options).toArray();
```

Comment:
With this approach, at every iteration, we'll need to execute a query to the database. I'm wondering if using the [query's cursor](https://mongodb.github.io/node-mongodb-native/api-generated/cursor.html) would be more efficient.

Essentially, the cursor controls retrieval of the documents in a way that is somewhat similar to the `Iterators` API, with _only one query_.

Reply:
Feel free to change it! I am not familiar with that and would welcome that change. :)

Reply:
Right! Will make an effort to try and get this into the release 😬